src/storage-controller/src/lib.rs

```diff
@@ -1812,11 +1807,17 @@ where
                         ingestions_to_drop.insert(*id);
                         source_statistics_to_drop.push(*id);
                     }
-                    DataSource::Progress => {
+                    DataSource::Progress | DataSource::Table { .. } | DataSource::Other => {
                         collections_to_drop.push(*id);
                     }
-                    DataSource::Other | DataSource::Introspection(_) => (),
-                    DataSource::Sink { .. } => {}
+                    DataSource::Introspection(_) | DataSource::Sink { .. } => {
+                        // Collections of these types are either not sources and should be dropped
+                        // through other means, or are sources but should never be dropped.
```

Comment:
This is based on my own opinion of what should be true and might be wrong! Please let me know and I'll remove the soft panic again.

Reply:
Sounds good if it passes!